### PR TITLE
Fix/svace issue 210209

### PR DESCRIPTION
**Pull request by lhs8928**
Commit 1: 454660
Add try catch statement to readModel

Commit 2: 455580
Add try catch statement to createDataset

Commit 3: 456427
Enclosed with #ifdef __ANDROID__, #endif cause if statement only reach on ANDROID environment

Commit 4: 456451
Initialized member variable expected_loss to 0.0

### CODE REVIEW

File: Applications/TransferLearning/CIFAR_Classification/jni/main.cpp [hunk 1]
```diff
@@ -418,7 +418,12 @@ int main(int argc, char *argv[]) {
     std::cerr << "Error during init" << std::endl;
     return 0;
   }
-  NN.readModel();
+  try {
+    NN.readModel();
+  } catch (...) {
+    std::cerr << "Error during readModel" << std::endl;
+    return 0;
```
Comment (kparichay) [inline Applications/TransferLearning/CIFAR_Classification/jni/main.cpp:425]: Please return a non-zero value as the failing error code.
0 is reserved for return on success.

File: Applications/TransferLearning/CIFAR_Classification/jni/main_func.cpp [hunk 2]
```diff
@@ -265,11 +265,16 @@ int main(int argc, char *argv[]) {
...
-                            getBatch_train);
-  dataset->setGeneratorFunc(ml::train::DatasetDataType::DATA_VAL, getBatch_val);
+  std::shared_ptr<ml::train::Dataset> dataset;
+  try {
+    dataset = createDataset(ml::train::DatasetType::GENERATOR);
+    dataset->setGeneratorFunc(ml::train::DatasetDataType::DATA_TRAIN,
+                              getBatch_train);
+    dataset->setGeneratorFunc(ml::train::DatasetDataType::DATA_VAL, getBatch_val);
+  } catch (...) {
+    std::cerr << "Error create dataset" << std::endl;
```
Comment (kparichay) [inline Applications/TransferLearning/CIFAR_Classification/jni/main_func.cpp:275]:
```suggestion
    std::cerr << "Error creating dataset" << std::endl;
```

File: Applications/TransferLearning/CIFAR_Classification/jni/main_func.cpp [hunk 3]
```diff
@@ -289,7 +289,12 @@ int main(int argc, char *argv[]) {
     std::cerr << "Error during init" << std::endl;
     return 0;
   }
-  model->readModel();
+  try {
+    model->readModel();
+  } catch (...) {
+    std::cerr << "Error during readModel" << std::endl;
+    return 0;
```
Comment (kparichay) [inline Applications/TransferLearning/CIFAR_Classification/jni/main_func.cpp:296]: return 1?

**Review by kparichay — COMMENTED** [review]
Also, one of the commits is missing?

File: Applications/TransferLearning/CIFAR_Classification/jni/main_func.cpp (hunk 2 above)
Comment (lhs8928) [reply Applications/TransferLearning/CIFAR_Classification/jni/main_func.cpp]: Done

File: Applications/TransferLearning/CIFAR_Classification/jni/main.cpp (hunk 1 above)
Comment (lhs8928) [reply Applications/TransferLearning/CIFAR_Classification/jni/main.cpp]: Done

File: Applications/TransferLearning/CIFAR_Classification/jni/main_func.cpp (hunk 3 above)
Comment (lhs8928) [reply Applications/TransferLearning/CIFAR_Classification/jni/main_func.cpp]: Done

**Comment by lhs8928** [comment]
> Also, one of the commits is missing? The description talks about two commits.

Yes.
In the process of fixing the previous commit I was missing the second commit
Now it is fixed.

**Review by kparichay — APPROVED** [review]
Please fix the clang CI issue
otherwise, LGTM

File: test/unittest/unittest_nntrainer_models.cpp [hunk 4]
```diff
@@ -327,6 +327,7 @@ GraphWatcher::GraphWatcher(const std::string &config, const bool opt) :
   }
 
   loss_node = NodeWatcher(graph.back());
+  expected_loss = 0.0;
```
Comment (kparichay) [inline test/unittest/unittest_nntrainer_models.cpp:330]: please set this in the initialization list - line 295.

Comment (lhs8928) [reply test/unittest/unittest_nntrainer_models.cpp]: Done

**Review by myungjoo — APPROVED** [review]
LGTM

**Pull request merged**